### ISSUE TITLE
Continuous integration

### ISSUE THREAD

**Issue by davidrpugh** [EconomicSL/housing-model#33]
@rafabap and @adrian-carro Is this repo being actively developed? If so, then I would like to start integrating the repo into the continuous integration infrastructure that I am developing.

**Comment by adrian-carro** [comment]
Yes, this repo is being actively developed and it will be more so in the months to come. What are the particular implications of your "continuous integration infrastructure"?

**Comment by davidrpugh** [comment]
@adrian-carro Continuous integration will perform that following checks for each commit to a PR against the important branches in each of our repos (typically `master` and `develop`)...

1. check that code compiles
2. check that unit tests pass
3. check that test coverage remains high
4. check that code quality remains high

...depending on our preferences we can also check that functional (or end-to-end) tests pass and the  there are no performance regressions.

**Comment by adrian-carro** [comment]
@davidrpugh can this issue be closed? I am fine with having "badges" as the codacy one in the readme file, but I can't invest time right now in unit testing or implementing all this integration infrastructure... should I close the issue and leave this for later on?

**Comment by davidrpugh** [comment]
@adrian-carro I am happy to implement the infrastructure for you, I was just waiting for the go ahead from you.  Actually writing tests is up to you.  IMHO you should take the time to come up with three core pieces of code or algorithms on which your results are likely to hinge and at least write tests for those.  Let's leave this issue open until I finish the CI infrastructure.

**Comment by adrian-carro** [comment]
Ok, no problem, I'll keep the issue open then.
